homeassistant/components/melcloud/climate.py

```diff
@@ -169,3 +209,128 @@ def max_temp(self) -> float:
         return convert_temperature(
             DEFAULT_MAX_TEMP, TEMP_CELSIUS, self.temperature_unit
         )
+
+
+ATW_ZONE_HVAC_MODE_LOOKUP = {
+    atw.ZONE_OPERATION_MODE_HEAT: HVAC_MODE_HEAT,
+    atw.ZONE_OPERATION_MODE_COOL: HVAC_MODE_COOL,
+}
+ATW_ZONE_HVAC_MODE_REVERSE_LOOKUP = {v: k for k, v in ATW_ZONE_HVAC_MODE_LOOKUP.items()}
+
+
+class AtwDeviceZoneClimate(MelCloudClimate):
+    """Air-to-Water zone climate device."""
+
+    def __init__(self, device: MelCloudDevice, atw_device: AtwDevice, atw_zone: Zone):
+        """Initialize the climate."""
+        super().__init__(device)
+        self._device = atw_device
+        self._zone = atw_zone
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return a unique ID."""
+        return f"{self.api.device.serial}-{self.api.device.mac}-{self._zone.zone_index}"
```

Comment:
Why would we do both serial and mac? Wouldn't just serial be unique enough to identify a device?

Reply:
Yes, it's a bit redundant. Using just the serial should be enough, but it would require either a breaking change for air-to-air devices or different behavior for air-to-air and air-to-water devices. What do you think?